src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDsa/ECDsaImportExport.cs

```diff
@@ -0,0 +1,347 @@
+                    Assert.True(parameters.Curve.IsPrime);
+                    var q = parameters.Q;
+                    var c = parameters.Curve;
+                    TestExplicitImportValidation(ec, ref q, ref c, true);
+
+                    q.X = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    q.X = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+                    q.X = new byte[1] { 0x10 }; TestExplicitImportValidation(ec, ref q, ref c);
+                    --q.X[0]; TestExplicitImportValidation(ec, ref q, ref c);
+
+                    q.Y = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    q.Y = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+                    q.Y = new byte[1] { 0x10 }; TestExplicitImportValidation(ec, ref q, ref c);
+                    --q.Y[0]; TestExplicitImportValidation(ec, ref q, ref c);
+
+                    c.A = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.A = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.A = new byte[1] { 0x10 }; TestExplicitImportValidation(ec, ref q, ref c);
+                    --c.A[0]; TestExplicitImportValidation(ec, ref q, ref c);
+
+                    c.B = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.B = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.B = new byte[1] { 0x10 }; TestExplicitImportValidation(ec, ref q, ref c);
+                    --c.B[0]; TestExplicitImportValidation(ec, ref q, ref c);
+
+                    c.Order = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.Order = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+
+                    c.Prime = null; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.Prime = new byte[] { }; TestExplicitImportValidation(ec, ref q, ref c);
+                    c.Prime = new byte[1] { 0x10 }; TestExplicitImportValidation(ec, ref q, ref c);
+                    --c.Prime[0]; TestExplicitImportValidation(ec, ref q, ref c);
+                }
+            }
+        }
+
+        private void TestExplicitImportValidation(ECDsa ec, ref ECPoint q, ref ECCurve c, bool success = false)
+        {
+            Assert.True(c.IsPrime);
+            ECParameters parameters = new ECParameters();
+            parameters.Q = q;
+            parameters.Curve = c;
+            if (success)
+            {
+                ec.ImportParameters(parameters);
+            }
+            else
+            {
+                Assert.ThrowsAny<CryptographicException>(() => ec.ImportParameters(parameters));
+            }
+
+            // Reset the values for the next test
+            parameters = ECDsaTestData.GetNistP256ExplicitTestData();
+            q = parameters.Q;
+            c = parameters.Curve;
```

Comment:
It seems weird that you're binding them to a particular curve, wouldn't this be more applicable if you just took the copy-from parameters struct as input?